sdks/cpp/common/examples/full_service/full_service.cpp

```diff
@@ -148,6 +172,12 @@ std::shared_ptr<grpc::ServerCredentials> getServerCredentials() {
         ssl_opts.pem_key_cert_pairs.push_back(
           grpc::SslServerCredentialsOptions::PemKeyCertPair{server_key, server_cert});
         ans = grpc::SslServerCredentials(ssl_opts);
+
+        if (absl::GetFlag(FLAGS_authz)) {
+            const std::shared_ptr<grpc::AuthMetadataProcessor> authzProcessor(new JWTAuthMetadataProcessor());
+            ans->SetAuthMetadataProcessor(authzProcessor);
```

Comment:
authzProcessor object goes out of scope line 179. investigate what SetAuthMetadataProcessor does with it - if it makes a copy, that's ok, if it's storing a reference, or pointer, that's a problem because it's now a dangling reference (i.e. a reference to a deleted object)

Reply:
authzProcessor is a std::shared_ptr passed by reference to SetAuthMetaDataProcessor(). It is hard to know exactly what grpc does with it behind the scene but I would assume that it makes some kind of copy of the shared_ptr. Given that the AuthMetaDataProcessor is running as expected I don't think this is an issue.

Reply:
yep - if it's anything like std::shared_ptr, it'll maintain a reference count and only delete the object once the final owning object is destroyed. This code is now considered safe ;-) Thanks for checking!